src/launcher/features/settings/RemoveArtifactoryTokenDialog.tsx

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright (c) 2025 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-4-Clause
+ */
+
+import React from 'react';
+import {
+    ConfirmationDialog,
+    describeError,
+    ErrorDialogActions,
+} from '@nordicsemiconductor/pc-nrfconnect-shared';
+
+import cleanIpcErrorMessage from '../../../common/cleanIpcErrorMessage';
+import { inMain as artifactoryToken } from '../../../ipc/artifactoryToken';
+import { useLauncherDispatch, useLauncherSelector } from '../../util/hooks';
+import { hasRestrictedAccessLevel } from '../sources/sourcesEffects';
+import { getSources } from '../sources/sourcesSlice';
+import {
+    getIsRemoveArtifactoryTokenVisible,
+    hideRemoveArtifactoryToken,
+    removeArtifactoryTokenInformation,
+} from './settingsSlice';
+
+export default () => {
+    const dispatch = useLauncherDispatch();
+    const isVisible = useLauncherSelector(getIsRemoveArtifactoryTokenVisible);
+    const nonPublicSources = useLauncherSelector(getSources).filter(source =>
+        hasRestrictedAccessLevel(source.url)
+    );
+
+    const hideDialog = () => dispatch(hideRemoveArtifactoryToken());
+
+    const removeToken = async () => {
+        try {
+            hideDialog();
+
+            await artifactoryToken.removeToken();
+            dispatch(removeArtifactoryTokenInformation());
+        } catch (error) {
+            dispatch(
+                ErrorDialogActions.showDialog(
+                    `Unable to remove token.`,
```

Comment:
```suggestion
                    `Unable to remove the token.`,
```

Reply:
Maybe @greg-fer should second this but I at least felt like he would have commented this and figured I might as well.